pitch.md

```diff
@@ -14,31 +23,170 @@ _What need is met by your idea? Who are you helping? What is that person's
+
+#### Geometric Representation
+
+We can also talk about the syntax of the internal geometric representation.
+
+The details of the syntax (actual names and symbols) are hard to pin down at
+this point, but we can talk about some of the concepts that we expect to come
+out in the syntax. For example, although Python, C++, and Java all have
+different surface syntax for constructing new objects, they all share the idea
+of the existence of classes and the existence of a function-like constructor
+that creates an instance of that class.
+
+The current image I have of the internal representation is that we need to be
+able to describe points and various types of point-sets (in particular lines
+and circles, maybe more). We need to be able to take intersections of point
+sets, choose points from point sets.
+
+I also think it would be cool (maybe even necessary) to be able to describe a
+construction as a sub-procedure. Perhaps we can define a construction which
+takes two points and gives you the midpoint of the segment between them.
+Perhaps the subprocedures can only take and return points? Perhaps they can
+also return point-sets? Perhaps point-sets shouldn't be fundamentals in the
+language?
+
+There are many questions which remain here.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+The internal representation should do three things during runtime.
+
+The first of these is consistency-checking. One could imagine syntactically
+correct statements about geometry which are not actually possible.
+
+The second of these is providing a usable representation of the geometric
+structure. This would be a way of communicating what has been drawn to some
+other system (Ex: the LaTeX rendering system).
+
+The third of these is providing a useful suggestion API. There should be some
+way of asking what actions could be performed next given the set of actions
+that have been performed so far and the set of macros / constructions which
+have been defined so far (/ are in scope?)
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+I want the following to feel seemless:
+   - Choosing points
+   - Drawing lines though them
+   - Drawing circles about them
+   - Finding the intesection of these loci
+   - Embedding constructions into other constructions concisely (subroutines)
+
+Fancy concepts should be definable using the above ideas, but not built in.
+The subroutines should feel like they we language primitives - true
+composability.
+
+I want it to be easy to search through possible actions using the internal
+geometric representation.
+
+I want it to be hard to move around individual points or lines. This is not the
+concern of someone doing a construction.
```

Comment:
But this could be important in making the output figure look a particular way, which might be important to the user. You could imagine a situation where the user is anal about making sure that the longest side of a construction is always oriented horizontally.